src/main/java/frc/robot/Commands/FollowTrajectory.java

```
@@ -0,0 +1,78 @@
+import edu.wpi.first.wpilibj2.command.RamseteCommand;
+import frc.robot.Subsystems.OdometryDrivetrain2;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+public class FollowTrajectory extends CommandBase {
+    private final OdometryDrivetrain2 drivetrain;
+    private Trajectory trajectory;
+    private RamseteCommand ramseteCommand;
+    private final RamseteController controller;
+    private DifferentialDriveKinematics kinematics;
+
+    public FollowTrajectory(OdometryDrivetrain2 drivetrain, Trajectory trajectory, PIDSettings leftPIDSettings,
+                            PIDSettings rightPIDSettings, FeedForwardSettings feedForwardSettings) {
+        this.drivetrain = drivetrain;
+        this.trajectory = trajectory;
+        controller = new RamseteController();
+        kinematics = new DifferentialDriveKinematics(drivetrain.getWidth());
+        ramseteCommand = new RamseteCommand(trajectory, drivetrain::getPose, controller,
+                new SimpleMotorFeedforward(feedForwardSettings.getkS(), feedForwardSettings.getkV(), feedForwardSettings.getkA()), kinematics,
+                drivetrain::getWheelSpeeds, new PIDController(leftPIDSettings.getkP(), leftPIDSettings.getkI(), leftPIDSettings.getkD()),
+                new PIDController(rightPIDSettings.getkP(), rightPIDSettings.getkI(), rightPIDSettings.getkD()), drivetrain::tankDriveVolts, drivetrain);
+    }
+
+    public FollowTrajectory(OdometryDrivetrain2 drivetrain, String file, PIDSettings leftPIDSettings,
+                            PIDSettings rightPIDSettings, FeedForwardSettings feedForwardSettings) {
+
+        this.drivetrain = drivetrain;
+        controller = new RamseteController();
+        kinematics = new DifferentialDriveKinematics(drivetrain.getWidth());
+        Path path = Path.of(Filesystem.getDeployDirectory().getPath(), file);
+        try {
+            trajectory = TrajectoryUtil.fromPathweaverJson(path);
+        } catch (IOException e) {
```

throw an exception instead of printing